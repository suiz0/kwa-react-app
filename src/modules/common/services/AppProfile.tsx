class AppProfile {
    isAuthenticated =  false;
    lang = 'en';
    langs: any[] = [];
    static Resources = {};
    client = "";
    isLoading = false;
  }

export default AppProfile;