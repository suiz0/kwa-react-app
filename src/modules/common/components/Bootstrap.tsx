import React from 'react';
import {load} from '../services/Config';
import AppProfile from '../services/AppProfile';

declare type callback = (props:any) =>any

interface BootstrapComponentProps {
    ready: callback;
    profile: AppProfile;
    config: any
}

class Bootstrap extends React.Component<BootstrapComponentProps, {config: any, isloading: boolean, profile: AppProfile}>
{
    constructor(props)
    {
        super(props);
        this.state = {
            config: this.props.config,
            isloading: true,
            profile: this.props.profile
        };
    }

    componentDidMount()
    {
        load()
        .then((config) => {
            const profile = {
                ...this.state.profile,
                client: config.client
            }
            this.setState({isloading: false, "config": config, "profile": profile});
        });
    }

    render()
    {
        // When configuration is loaded executes ready method
        if(!this.state.isloading)
        {
            return this.props.ready({config:this.state.config, profile: this.state.profile});
        }

        // While configuration is loaded prints Loading "indicator"
        return <div>Loading...</div>;
    }
}

export default Bootstrap;