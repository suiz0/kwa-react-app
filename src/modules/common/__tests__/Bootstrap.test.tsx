import React from 'react';
import Bootstrap from '../components/Bootstrap';
import AppProfile from '../services/AppProfile';
import Adapter from 'enzyme-adapter-react-16';
import {shallow, configure} from 'enzyme';

configure({adapter: new Adapter()});

global.fetch = jest.fn().mockImplementation(()=> {
    return Promise.resolve({json: ()=>Promise.resolve({})});
});

let component;
const options = {
    ready: jest.fn(),
    profile: new AppProfile(),
    config: {}
};

beforeEach(()=>{
    component = shallow(<Bootstrap {...options} />);
});

describe("Test Rendering", () => {
    

    test('renders without crashing', () => {
        expect(component).toMatchSnapshot();
    });
});

describe('Test behaviors', ()=> {
    test('executes ready method once config is loaded', ()=>{
       expect(options.ready).toBeCalled();
    });
});
