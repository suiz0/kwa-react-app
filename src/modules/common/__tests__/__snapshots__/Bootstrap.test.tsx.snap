// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Rendering renders without crashing 1`] = `
<div>
  Loading...
</div>
`;
